### DISCUSSION TITLE
Embedded db outside of react app

### DISCUSSION THREAD

**Discussion by jbellmore** [keystonejs/keystone#7891] [Questions]
Hi all!

I am just getting started with KeystoneJS and really impressed so far. For the use case I am trying it out for I am following the walkthrough for using the new Embedded mode with SQLite:

https://keystonejs.com/docs/walkthroughs/embedded-mode-with-sqlite-nextjs#standalone-vs-embedded-modes

Everything works great and I have the app and the admin UI started up and working properly. The question I have is, is there a way to use Keystone outside of the React app in order to populate the database? I have some other node code that is parsing through some various data sources that I would like to populate the SQLite database from.

Essentially I am building a standalone offline UI for connecting and visualizing various data sources. I would love to be able to be able to import 'query' like in the react pages, but from outside of the react app (a node cli or something).

Is there any reasonable way to do this at the moment?

Thanks!

**Comment by flexdinesh** [comment]
You can run queries to import your data in a few ways:

1. You could extend the GraphQL schema to write your own mutation that takes care of this import you're talking about. You could invoke this mutation via the GraphQL API or in our `db.onConnect` or in a scheduler.
2. You could import data when your server starts using a script argument in your `db.onConnect` method. [Example here](https://github.com/keystonejs/keystone/blob/8ba5a3526b0f99d1c648d95b49cfcac80d7b6a9a/examples/document-field-customisation/keystone-server/keystone.ts#L12).
